commands/announce.lua

```diff
@@ -0,0 +1,81 @@
+local Utils = require('utils.utils')
+local Core = require('utils.core')
+local Public = {}
+
+---Spawns announcement speech-bubble. It's supposed to be called on every
+---map reset and on-demand.
+Public.announce_if_any = function()
+    local announcement = storage.announcement
+    if not announcement.text then
+        return
+    end
+
+    ---@type LuaSurface
+    local s = game.surfaces[storage.bb_surface_name]
+    local e = s.create_entity({
+        name = 'steel-chest',
+        position = { 0, -9 },
+        force = 'neutral',
+        minable_flag = false,
+        operable = false,
+        destructible = false,
+    })
+
+    s.create_entity({
+        name = 'compi-speech-bubble',
+        text = announcement.text,
+        source = e,
+        position = e.position,
+    })
+
+    announcement.entity = e
+end
+
+---Implements /announce command.
+---If no parameter is set, it will clear existing announcement if any.
+---If parameter is set, it will create new announcement or overwrite existing one.
+---The announcement is supposed to persist across map resets.
+---@param cmd CustomCommandData
+local function announce(cmd)
+    if not cmd.player_index then
+        return
+    end
+
+    local player = game.players[cmd.player_index]
+    if not player or not player.valid then
+        return
+    end
+
+    if not player.admin then
+        player.print('This command can only be used by admins')
+        return
+    end
+
+    -- In any case, we're going to clear existing announcement if any.
+    local removed = false
+    local bubble = storage.announcement.entity
+    if bubble and bubble.valid then
+        removed = true
+        bubble.destroy()
```

Comment:
(Optional)
Check if the removal actually succeeded?
```suggestion
        removed = bubble.destroy()
```

Reply:
I'm not sure, just noticed in API that `destroy()` returns bool

Reply:
Done.